smithy-dafny-conversion/src/test/java/software/amazon/smithy/dafny/conversion/ToNativeTest.java

```diff
@@ -105,4 +106,26 @@ public void testGenericToMap() {
     Map<String, String> actual = MapStringString(input);
     assertEquals(expected, actual);
   }
+
+  @Test
+  public void testGenericToMapInteger() {
+    Map<String, Integer> expected = new LinkedHashMap<>();
+    expected.put("one", 1);
+    expected.put("two", 2);
+    Map<DafnySequence<? extends Character>, Integer> temp =
+      new LinkedHashMap<>();
+    temp.put(DafnySequence.asString("one"), 1);
+    temp.put(DafnySequence.asString("two"), 2);
+    DafnyMap<DafnySequence<? extends Character>, ? extends Integer> input =
+      new DafnyMap<>(temp);
+    // This conversion would not work in a smithy-dafny project,
```

Comment:
Without context the statement makes me nervous, I would point specifically to the generics issue

Reply:
```suggestion
    // This conversion would not work in a smithy-dafny project,
    // because it does not use the concrete type `Integer`
```